dashboard/app/helpers/levels_helper.rb

```diff
@@ -179,15 +179,22 @@ def app_options
     # Unsafe to generate these twice, so use the cached version if it exists.
     return @app_options unless @app_options.nil?
 
-    if (@level.channel_backed? && params[:action] != 'edit_blocks') || @level.is_a?(Javalab)
+    should_load_channel = (@level.channel_backed? && params[:action] != 'edit_blocks') || @level.is_a?(Javalab)
+    # If we have loaded a user, the page must not be publicly cached
+    level_not_cached = current_user || @user
```

Comment:
What do you think about changing the logic here to use or match [@public_caching](https://github.com/code-dot-org/code-dot-org/blob/226135b9104428009bf9abb4a45ed2001020bb6c/dashboard/app/helpers/levels_helper.rb#L232-L233)?  It does change the behavior somewhat -- signed out users on non-cached levels would go through the exisitng flow (which I think is good?).

Reply:
I didn't realize there was logic I could leverage other than checking for a user, this is a way better option!